relayer/src/foreign_client.rs

```diff
@@ -1067,17 +1061,32 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
     /// Retrieves all consensus states for this client and sorts them in descending height
     /// order. If consensus states are not pruned on chain, then last consensus state is the one
     /// installed by the `CreateClient` operation.
-    fn consensus_states(&self) -> Result<Vec<AnyConsensusStateWithHeight>, ForeignClientError> {
+    ///
+    /// The optional `u64` can be used to limit the number of consensus states that
+    /// are requested.
+    fn consensus_states(
```

Comment:
What do you think about renaming this method to reflect the fact that it now can fetch the provided number of consensus states? Perhaps `fetch_consensus_states_with_limit` or something like that?

Reply:
8999ff55